pkgs/by-name/bi/bitwarden-directory-connector/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  pkgs,
+  git,
+  python3,
+  pkg-config,
+  libsecret,
+  nodejs_18,
+}: let
+  buildNpmPackage' = buildNpmPackage.override {nodejs = nodejs_18;};
+in
+  buildNpmPackage' rec {
+    pname = "bitwarden-directory-connector";
+    version = "2023.10.0";
+
+    src = fetchFromGitHub {
+      owner = "bitwarden";
+      repo = "directory-connector";
+      rev = "v${version}";
+      hash = "sha256-CgaCnMWNVWCJBypNcdoseVCwD8Mlq4YaWpK+VZT/7Qk=";
+      leaveDotGit = true;
+    };
+
+    npmDepsHash = "sha256-jBAWWY12qeX2EDhUvT3TQpnQvYXRsIilRrXGpVzxYvw=";
+
+    env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+    makeCacheWritable = true;
+    npmBuildScript = "build:cli:prod";
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out
+      cp -R {build-cli,node_modules} $out
+      runHook postInstall
```

Comment:
I think it would be a good idea to create a wrapper for `${lib.getExe nodejs} ${connector}/build-cli/bwdc.js` here and set `meta.mainProgram`. Then you can simply refer to `${lib.getExe connector}` in the scripts above.

Reply:
Would you know of an existing program I cna reference for this?

Reply:
Yeah I can send an example later. You can also grep nixpkgs for `wrapProgram.+node` to find some examples.

Reply:
Am I right in thinking that it should go n the package side of things?  
That it basiclaly turns it into a standalone "binary" that ye cna just run?

Reply:
> That it basiclaly turns it into a standalone \"binary\" that ye cna just run?

Yep. You then need to set mainProgram to the name of the wrapper in $out/bin for lib.getExe to work.

Reply:
got it to work, thanks for pointing me to it.